### PR TITLE
Added Settings Fragment as Preferences.

### PR DESCRIPTION
**Pull request by jddeep**
Screenshot of the SettingsFragment -
firstpullreqneurolab
Fixes #[17]

**Changes**: [Add here what changes were made in this issue and if possible provide links.]

**Screenshot/s for the changes**: [Add screenshot/s of the layout where you made changes or a `*.gif` containing a demonstration]

**Checklist**: [Please tick following check boxes with `[x]` if the respective task is completed]
- [ ] I have used resources from `strings.xml`, `dimens.xml` and `colors.xml` without hard-coding them
- [ ] No modifications done at the end of resource files `strings.xml`, `dimens.xml` or `colors.xml`
- [ ] I have reformatted code in every file included in this PR [<kbd>CTRL</kbd>+<kbd>ALT</kbd>+<kbd>L</kbd>]
- [ ] My code does not contain any extra lines or extra spaces
- [ ] I have requested reviews from other members

**APK for testing**: [Compress the `app-debug.apk` file into a `<feature>.rar` or `<feature>.zip` file and upload it here]

### CODE REVIEW

**Comment by mariobehling** [comment]
Thank you. Please add issue number and ensure Travis builds.

File: app/build.gradle [hunk 1]
```diff
@@ -27,7 +27,9 @@ dependencies {
     implementation "com.android.support:appcompat-v7:$rootProject.supportLibraryVersion"
     implementation "com.android.support:design:$rootProject.supportLibraryVersion"
     implementation "com.android.support.constraint:constraint-layout:$rootProject.constraintLayoutVersion"
+    implementation 'com.android.support:preference-v7:26.1.0'
     // Test
+    implementation 'com.android.support.constraint:constraint-layout:1.1.3'
```
Comment (CloudyPadmal) [inline app/build.gradle:32]: This is added in line 29

File: app/build.gradle [hunk 2]
```diff
@@ -27,7 +27,9 @@ dependencies {
     implementation "com.android.support:appcompat-v7:$rootProject.supportLibraryVersion"
     implementation "com.android.support:design:$rootProject.supportLibraryVersion"
     implementation "com.android.support.constraint:constraint-layout:$rootProject.constraintLayoutVersion"
+    implementation 'com.android.support:preference-v7:26.1.0'
```
Comment (CloudyPadmal) [inline app/build.gradle:30]: Use the same library version across the whole project. Change this to;
`implementation "com.android.support:preference-v7:$rootProject.supportLibraryVersion"`

File: app/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -19,6 +19,7 @@
                 <category android:name="android.intent.category.LAUNCHER" />
             </intent-filter>
         </activity>
+        <activity android:name=".BCISettingsActivity"></activity>
```
Comment (CloudyPadmal) [inline app/src/main/AndroidManifest.xml:22]: This is not BCI. Change **BCI** to **Neuro**

File: app/src/main/java/io/neurolab/BCISettings.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
+package io.neurolab;
+
+
+import android.os.Build;
+import android.support.annotation.RequiresApi;
+
+import java.util.concurrent.ConcurrentLinkedDeque;
+
+@RequiresApi(api = Build.VERSION_CODES.LOLLIPOP)
```
Comment (CloudyPadmal) [inline app/src/main/java/io/neurolab/BCISettings.java:9]: We need to support from API level 16

File: app/src/main/java/io/neurolab/BCISettings.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+public class BCISettings {
+
+    protected ConcurrentLinkedDeque<double[]> currentData = new ConcurrentLinkedDeque<double[]>();
+    public int samplesPerSecond = 3;
+    public int numChannels = 2;
+    public int bins = 4;
+
+    //@TODO
```
Comment (CloudyPadmal) [inline app/src/main/java/io/neurolab/BCISettings.java:17]: Remove

File: app/src/main/java/io/neurolab/BCISettings.java [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+    public BCISettings() {
+    }
+
+    public BCISettings(int samplesPerSecond, int numChannels, int bins ) {
+
```
Comment (CloudyPadmal) [inline app/src/main/java/io/neurolab/BCISettings.java:22]: Remove excess spacing

File: app/src/main/res/values/styles.xml [hunk 7]
```diff
@@ -17,4 +18,6 @@
 
     <style name="AppTheme.PopupOverlay" parent="ThemeOverlay.AppCompat.Light" />
 
+
```
Comment (CloudyPadmal) [inline app/src/main/res/values/styles.xml:21]: Please do not add extra new lines like this

File: app/src/main/res/xml/bci_settings_fragment.xml [hunk 8]
```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0" encoding="utf-8"?>
+<PreferenceScreen xmlns:android="http://schemas.android.com/apk/res/android">
+
+    <EditTextPreference
+        android:defaultValue="1"
+        android:title="Samples per second"
+        android:key="samples"
+        />
```
Comment (CloudyPadmal) [inline app/src/main/res/xml/bci_settings_fragment.xml:8]: Reformat the code using [<kbd>CTRL</kbd>+<kbd>ALT</kbd>+<kbd>L</kbd>], Keep the `/>` at the end of the last parameter instead on a new line.

**Review by CloudyPadmal — CHANGES_REQUESTED** [review]
Thank you for the PR. Please change the name `BCI` to `Neuro` in every instance. Reformat the code and remove all the extra new lines and spaces added.

**Comment by jddeep** [comment]
There were some better changes to be made to resolve the issue which I understood later. I will make the corrected pull request again.

**Comment by jddeep** [comment]
There were some better changes to be made to resolve the issue which I understood later. I will make the corrected pull request again.

**Comment by CloudyPadmal** [comment]
Please don't close PRs and open another one with the changes. We need to follow up on the progress. Similar issues I have reviewed here still exist in that PR. Now I have to comment all over again on a brand new PR. Please double check first.

Those issues you have opened #20 #21 are issues in your PR, not in the main repository. No need to open issues like that.

**Comment by jddeep** [comment]
Okay I will make sure about that @CloudyPadmal  from next time. But rest assure i closed the issues #20 and #21 and made all the changes that were required and now the new PR is completely mergeable.

**Pull request closed without merging**